File: go/api/MLog.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// MLog 对应 MLog.py。
type MLog struct {
	*session.ApiGroup
}

// NewMLog 创建 MLog API 分组。
func NewMLog(s *session.Session) *MLog {
	return &MLog{ApiGroup: session.NewApiGroup(s)}
}

// MobileLogger 对应 mlog/mobile-logger。
func (m *MLog) MobileLogger() (map[string]interface{}, error) {
	return m.S.Get("mlog/mobile-logger", nil, "api")
}

File: src/api/m_log.rs
//! MLog API group (`api/MLog.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// MLog API group.
pub struct MLog<'a> {
    conn: &'a Connection,
}

impl<'a> MLog<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        MLog { conn }
    }

    /// `mlog/mobile-logger`. Mobile logger information (reverse engineered).
    pub fn mobile_logger(&self) -> Result<Value> {
        get_value(self.conn, "api/mlog/mobile-logger")
    }
}
